src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -67,6 +67,114 @@ public static void Resize<T>([NotNull] ref T[]? array, int newSize)
             Debug.Assert(array != null);
         }
 
+        [RequiresDynamicCode("The native code for the array might not be available at runtime.")]
+        public static unsafe Array CreateInstance(Type elementType, int length)
+        {
+            if (elementType is null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.elementType);
+            if (length < 0)
+                ThrowHelper.ThrowLengthArgumentOutOfRange_ArgumentOutOfRange_NeedNonNegNum();
+
+            RuntimeType? t = elementType.UnderlyingSystemType as RuntimeType;
+            if (t == null)
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Arg_MustBeType, ExceptionArgument.elementType);
+
+            return InternalCreate(t, 1, &length, null);
+        }
+
+        public static unsafe Array CreateInstance(Type elementType, int length1, int length2)
```

Comment:
Do we need `[RequiresDynamicCode]` on length2 and length3 overloads?

Reply:
I know you are moving code around but will developers understand what "MDArrays of Rank != 1 can be created because they don't implement generic interfaces"  means?

Reply:
This is not user facing message. This message is a comment for the few developers on .NET runtime team who are going to maintain the array implementation. I think it is sufficient for those.

If you would like to see this message to be worded differently, please add it as a suggestion.